File: Sorting/bubble_sort.cpp
/**
 * @Author: hardcodecoder
 * @Date:   02:48:32  Tuesday 24 March 2020
 * @Last modified by:   hardcodecoder
 * @Last modified time: 03:19:13  Tuesday 24 March 2020
 */


#include<stdio.h>
#include<stdlib.h>


//Performs bubble sort on the array
void bubbleSort(int *a, int n) {
  int i, j;
  for (i = 0; i < n-1; i++){
    //Iterate only upto n-i-1 time
    //since the largest value get's pushed tothe last index
    //at every pass and needs no more comparision
    for (j = 0; j < n-i-1; j++){
      //Swaps two consecutive array values
      //If next index has a smaller value than current index
      if (a[j] > a[j+1]) {
        int t = a[j];
        a[j] = a[j+1];
        a[j+1] = t;
      }
    }
  }
}


//Driver method
int main() {
  int *a;
  int i, n;

  printf("Enter the size of the array : ");
  scanf("%d", &n);

  a=(int*)malloc(sizeof(int)*n);

  printf("\nEnter the data: ");
  for (i = 0; i < n; i++)
    scanf("%d", &a[i]);

  printf("\nUnsorted data: ");
  for (i = 0; i < n; i++)
    printf("%d ", a[i]);

  bubbleSort(a, n);

  printf("\nSorted data: ");
  for (i = 0; i < n; i++)
    printf("%d ", a[i]);

  return 0;
}

File: Sorting/Java/BubbleSort.java
/**
 * @Author: hardcodecoder
 * @Date:   03:43:46 Friday 15 May 2020
 * @Last modified by:   hardcodecoder
 * @Last modified time: 03:59:38 Friday 15 May 2020
 */


import java.io.*;
import java.util.Scanner;

public class BubbleSort {

  public static void main(String args[]) {
    int n;
    System.out.println("Enter the number of elements: ");
    Scanner sc = new Scanner(System.in);
    n = sc.nextInt();

    float[] list = new float[n];

    System.out.println("Enter the numbers in the arrray:");
    for (int i = 0; i < n; i++)
      list[i] = sc.nextFloat();

    bubbleSort(list);

    System.out.print("Sorted array: ");
    for (int i = 0; i < n; i++)
      System.out.print(String.format("%.2f, ", list[i]));
    System.out.println();
  }

  private static void bubbleSort(float[] list) {
    int n = list.length;
    for (int i = 0; i < n-1; i++) {
      for (int j = 0; j < n-1-i; j++) {
        if (list[j] > list[j+1]) {
          // Value at jth index is smaller than (j+1)th index
          // swap them to maintain ascending order
          float s = list[j];
          list[j] = list[j+1];
          list[j+1] = s;
        }
      }
    }
  }
}
